File: test22_out/atmel.atmega328/Mcu/Mcu.js

/*
 * ======== MODULE atmel.atmega328.Mcu ========
 */

var $$c = function() {
    const Mcu = {};
    Mcu.Mcu = Mcu
    Mcu.$name = 'atmel.atmega328.Mcu';
    Mcu.pollen$used = 0;
    Mcu.reset = new $$Ref('atmel_atmega328_Mcu_reset__E');
    Mcu.shutdown = new $$Ref('atmel_atmega328_Mcu_shutdown__E');
    Mcu.wait = new $$Ref('atmel_atmega328_Mcu_wait__E');
    Mcu.cycle = new $$Ref('atmel_atmega328_Mcu_cycle__E');
    Mcu.getFrequency = new $$Ref('atmel_atmega328_Mcu_getFrequency__E');
    Mcu.setFrequency = new $$Ref('atmel_atmega328_Mcu_setFrequency__E');
    Mcu.targetInit = new $$Ref('atmel_atmega328_Mcu_targetInit__I');
    Mcu.getFrequencyOnHost = function() {
        var $$text = '';
        return(  (Mcu.sysFreq)  );
        return $$text;
    }
    Mcu.setFrequencyOnHost = function( hz ) {
        var $$text = '';
        (Mcu.sysFreq) = (hz);
        return $$text;
    }
    Mcu.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    Mcu.pollen__uses__ = function() {
    }
    Mcu.$$privateInit = function() {
        Mcu.sysFreq = 0;
    }
    return Mcu;
}
var $$u = $$c();
$units['atmel.atmega328.Mcu'] = $$u;
$units.push($$u);

File: test22_out/atmel.atmega328/Mcu/Mcu.c

/*
 * ======== string literals ========
 */

static const char atmel_atmega328_Mcu__s__fileName[] = "Mcu.p";
static const char atmel_atmega328_Mcu__s__0[] = "Mcu";

/*
 * ======== function definitions ========
 */

void atmel_atmega328_Mcu_wait__E( uint32 us ) {
    for (; (us) > 0; --(us)) {
         asm ("nop") ;

         asm ("nop") ;

         asm ("nop") ;

         asm ("nop") ;

         asm ("nop") ;

         asm ("nop") ;

         asm ("nop") ;

         asm ("nop") ;

         asm ("nop") ;

         asm ("nop") ;

    }
}

uint32 atmel_atmega328_Mcu_getFrequency__E() {
    return(  atmel_atmega328_Mcu_sysFreq__V  );
}

void atmel_atmega328_Mcu_reset__E() {
}

void atmel_atmega328_Mcu_cycle__E() {
     asm ("nop") ;

}

void atmel_atmega328_Mcu_targetInit__I() {
}

void atmel_atmega328_Mcu_shutdown__E() {
}

void atmel_atmega328_Mcu_setFrequency__E( uint32 hz ) {
    atmel_atmega328_Mcu_sysFreq__V = (hz);
}

